File: src/main/java/me/jasper/jasperproject/JasperItem/Util/Factory.java
package me.jasper.jasperproject.JasperItem.Util;

import me.jasper.jasperproject.JasperItem.JItem;
import me.jasper.jasperproject.JasperProject;

import java.io.ObjectStreamClass;
import java.io.Serializable;

public interface Factory extends Serializable {
    JItem create();

    default JItem finish(){
        JItem product = create();
        long serialVersionUID = ObjectStreamClass.lookup(this.getClass()).getSerialVersionUID();
        product.setVersion(serialVersionUID);
        product.update();
        return product;
    }
}

File: src/main/java/me/jasper/jasperproject/JMinecraft/Item/Util/Factory.kt
package me.jasper.jasperproject.JMinecraft.Item.Util

import me.jasper.jasperproject.JMinecraft.Item.JItem
import java.io.ObjectStreamClass
import java.io.Serializable

interface Factory : Serializable {
    fun create(): JItem

    fun finish(): JItem {
        val product = create()
        val serialVersionUID = ObjectStreamClass.lookup(this.javaClass).serialVersionUID
        product.version = serialVersionUID
        product.update()
        return product
    }
}
